MetaMask/metamask-mobile
# feat: display snap name

## Description

This PR updates the display of snap accounts in the MetaMask mobile app. Previously, all snap accounts were shown with a generic “Snaps (Beta)” label. This change replaces the generic label with the specific name of the snap when available, providing users with clearer and more useful information about their snap accounts.

Key changes include:

- Creation of the getInternalAccountByAddress utility function to retrieve account information of the InternalAccount type.
- Modification of getAccountLabelTextByKeyring to return the snap name for snap accounts, or “Snaps (Beta)” if the name is unavailable.
- Updates to affected components to display the correct snap name in the account list.
- Unit tests added to verify the new functionality.

## Related issues

Fixes: https://github.com/MetaMask/accounts-planning/issues/603

## Manual testing steps

1. Build the mobile client in Flask
2. Navigate to the [SSK dapp](https://metamask.github.io/snap-simple-keyring/latest/), install the snap, and create a snap account
3. Open the account list
4. You _should_ see a tag under your new snap account that says "MetaMask Simple Snap Keyring"
5. Note: we do not have any snaps yet that do not have names, so we will need to rely on the jest unit tests for the "Snaps (Beta)" rendering behavior 

## Screenshots/Recordings

**Before**

<image src="https://github.com/user-attachments/assets/1c34187c-3807-4514-a4e2-55a6d60b1e1c" width="350" />

<image src="https://github.com/user-attachments/assets/72c2950a-5c3a-444d-939d-c82aa615457e" width="350" />


**After**

<image src="https://github.com/user-attachments/assets/23c93404-097f-4736-8fa8-f11fa69f3de5" width="350" />

<image src="https://github.com/user-attachments/assets/b5cd66a6-4c0e-4dc9-9d6e-528f675041c1" width="350" />


Pre-merge author checklist

- [x] I’ve included unit tests to cover the new behavior.
- [x] I’ve applied the correct labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).

Pre-merge reviewer checklist

- [ ] I’ve manually tested the PR by running the app and testing the modified behavior.
- [ ] I confirm that this PR addresses all acceptance criteria described in the issue and includes the necessary testing evidence such as screenshots and recordings.
